### PR TITLE
fix streams calculation issue for latency mode with big threads input

### PR DESCRIPTION
**Pull request by wangleis**
### Details:
 - *fix streams calculation issue for latency mode with big threads input*
 - *refactor streams calculation function and add more test data for all code paths*

### Tickets:
 - *[issues 21354](https://github.com/openvinotoolkit/openvino/issues/21354)*

### CODE REVIEW

File: src/plugins/intel_cpu/src/cpu_streams_calculation.cpp [hunk 1]
```diff
@@ -146,68 +146,63 @@ std::vector<std::vector<int>> get_streams_info_table(const int input_streams,
...
+                if ((n_threads_per_stream > proc_type_table[0][MAIN_CORE_PROC]) &&
+                    (proc_type_table[0][MAIN_CORE_PROC] > 0)) {
+                    stream_info[PROC_TYPE] = ALL_PROC;
+                }
+            }
+        } else if (((input_streams_changed == false) &&
+                    (latencyThreadingMode == Config::LatencyThreadingMode::PER_PLATFORM)) ||
+                   (proc_type_table.size() == 1) || ((input_streams_changed == true) && (input_streams == 1))) {
+            n_streams = 1;
+            n_threads_per_stream = std::min(input_threads, proc_type_table[0][ALL_PROC]);
```
Comment (xufang-lisa) [inline src/plugins/intel_cpu/src/cpu_streams_calculation.cpp:164]: here input_threads == 0, right?

File: src/plugins/intel_cpu/src/cpu_streams_calculation.cpp [hunk 2]
```diff
@@ -146,68 +146,63 @@ std::vector<std::vector<int>> get_streams_info_table(const int input_streams,
...
+                    stream_info[THREADS_PER_STREAM] = n_threads_per_stream;
+                    update_ids_method(proc_type_table[0]);
+                } else if (proc_type_table[0][MAIN_CORE_PROC] == 0) {
+                    stream_info[PROC_TYPE] = EFFICIENT_CORE_PROC;
+                    n_threads_per_stream = proc_type_table[0][EFFICIENT_CORE_PROC];
+                    stream_info[THREADS_PER_STREAM] = n_threads_per_stream;
+                    update_ids_method(proc_type_table[0]);
+                } else {
+                    stream_info[PROC_TYPE] = ALL_PROC;
+                    n_threads_per_stream = proc_type_table[0][ALL_PROC];
```
Comment (sunxiaoxia2022) [inline src/plugins/intel_cpu/src/cpu_streams_calculation.cpp:181]: If model_prefer_threads == proc_type_table[0][MAIN_CORE_PROC] + proc_type_table[0][EFFICIENT_CORE_PROC], all cores would be used?

File: src/plugins/intel_cpu/src/cpu_streams_calculation.cpp (hunk 1 above)
Comment (wangleis) [reply src/plugins/intel_cpu/src/cpu_streams_calculation.cpp]: Updated.

File: src/plugins/intel_cpu/src/cpu_streams_calculation.cpp (hunk 2 above)
Comment (wangleis) [reply src/plugins/intel_cpu/src/cpu_streams_calculation.cpp:180]: Yes. latency mode support using hyper threads when user set `enable_hyper_threading`.

`proc_type_table[0][HYPER_THREADING_PROC]` is updated by `enable_hyper_threading` in `apply_hyper_threading()`

**Review by xufang-lisa — APPROVED** [review]

**Review by sunxiaoxia2022 — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**